src/foundry/foundry.js/pixi/points/wallEndpoint.d.ts

```diff
@@ -0,0 +1,88 @@
+/**
+ * A special subclass of PIXI.Point which is used for modeling Wall endpoints.
+ * A wall endpoint must have integer coordinates.
+ *
+ * This was used for the RadialSweepPolygon but can now be deleted once that is
+ * @deprecated since v9d2
+ */
+declare class WallEndpoint extends PIXI.Point {
+  /**
+   * @param x - The integer x-coordinate
+   * @param y - The integer y-coordinate
+   */
+  constructor(x: number, y: number);
+
+  /**
+   * Express the point as a 32-bit integer with 16 bits allocated to x and 16 bits allocated to y
+   */
+  key: number;
+
+  /**
+   * The angle between this point and the polygon origin
+   * @defaultValue `undefined`
+   */
+  angle: number | undefined;
+
+  /**
+   * Record the set of walls which connect to this Endpoint
+   */
+  walls: Set<Wall>;
+
+  /**
+   * Record whether this point is the endpoint of any Wall
+   * @defaultValue `false`
+   */
+  isEndpoint: boolean;
+
+  /**
+   * Record whether this point is a midpoint of any wall?
+   * @defaultValue `false`
+   */
+  isMidpoint: boolean;
+
+  /**
+   * Record whether this point is the termination of the Ray
+   * @defaultValue `false`
+   */
+  isTerminal: boolean;
+
+  /**
+   * Aggregate the maximum of each wall restriction type
+   */
+  types: Record<foundry.CONST.WALL_RESTRICTION_TYPES, number>;
+
+  /**
+   * An intermediate variable used to store the proportional distance of this point from a SightRay origin
+   * @defaultValue `undefined`
+   */
+  protected _r: number | undefined;
+
+  /**
+   * An intermediate variable used to cache the continuation attributes for a certain point
+   * @defaultValue `undefined`
+   */
```

Comment:
```suggestion
   * @defaultValue `undefined`
   * @internal
   */
```

Reply:
This wasn't marked as `@private` in the source, so I left it merely as `protected`.